flooder/src/main.rs

```diff
@@ -1,100 +1,258 @@
+        let pending_in_pool = connections[0].pending_extrinsics_len().await?;
+        overdue_transactions += schedule.transactions_in_interval;
+        let transactions_to_pool_limit = pool_limit.saturating_sub(pending_in_pool);
+        let transactions_to_send = min(transactions_to_pool_limit, overdue_transactions);
+        overdue_transactions -= transactions_to_send;
+        info!(
+            "Interval {}: submitting {} txns",
+            i + 1,
+            transactions_to_send
+        );
+        debug!("In txpool there are {pending_in_pool} txns. Overdue txns: {overdue_transactions}");
 
-                    let left_duration = Duration::from_secs(time).saturating_sub(dur);
+        let tasks = connections
+            .iter()
+            .enumerate()
+            .map(|(conn_id, conn)| {
+                let mut per_task = transactions_to_send / connections.len() as u64;
+                if (conn_id as u64) < transactions_to_send % connections.len() as u64 {
+                    per_task += 1;
+                }
 
-                    info!("sleeping for {}ms", left_duration.as_millis());
-                    sleep(left_duration).await;
+                let dest = dest.clone();
+                let nonce = nonces[conn_id];
+                nonces[conn_id] += per_task as u32;
+                async move {
+                    for tx_id in 0..per_task as u32 {
+                        conn.sign_with_params(
+                            transfer_keep_alive(dest.clone(), transfer_amount),
+                            Default::default(),
+                            nonce + tx_id,
+                        )?
+                        .submit(status)
+                        .await?;
+                    }
+                    anyhow::Ok(())
                 }
             })
-        })
-        .collect();
+            .collect::<Vec<_>>();
 
-    join_all(handles).await;
+        for result in join_all(tasks).await {
+            if let Err(e) = result {
+                info!("Error submitting transaction: {e:?}");
+                return Err(e);
+            }
+        }
+
+        let duration = Instant::now().saturating_duration_since(start);
+        if duration.as_secs() >= schedule.interval_duration * schedule.intervals {
+            break;
+        }
+        let left_duration =
+            Duration::from_secs((i + 1) * schedule.interval_duration).saturating_sub(duration);
+        sleep(left_duration).await;
+    }
+
+    info!("Flooding time has passed, left {overdue_transactions} overdue txns because of the pool limit.");
+    if return_balance_at_the_end {
+        for (conn_id, conn) in connections.iter().enumerate() {
+            conn.sign_with_params(
+                transfer_all(dest.clone(), true),
+                Default::default(),
+                nonces[conn_id],
+            )?
+            .submit(status)
+            .await?;
+        }
+        debug!("Returned balance back to main account");
+    }
```

Comment:
this should be moved out of flood fn - this is not related to this function

Reply:
Moved, required returning nonces. Alternatively we could submit a tx with current nonce, but higher tip, but I guess some race conditions could occur that way.